rust/foxglove/Cargo.toml

```diff
@@ -55,6 +55,7 @@ tokio-tungstenite = { workspace = true, optional = true }
 tokio-util = { workspace = true, optional = true }
 tokio = { workspace = true, optional = true }
 tracing.workspace = true
+urlencoding = "2.1.3"
```

Comment:
There's also this crate: https://crates.io/crates/percent-encoding

They both seem popular and neither have dependencies. I think there's no reason to change this choice.